superset/db_engine_specs/base.py

```diff
@@ -886,12 +891,18 @@ def get_sqla_column_type(cls, type_: str) -> Optional[TypeEngine]:
         """
         Return a sqlalchemy native column type that corresponds to the column type
         defined in the data source (return None to use default type inferred by
-        SQLAlchemy). Needs to be overridden if column requires special handling
+        SQLAlchemy). Override `_column_type_mappings` for specific needs
         (see MSSQL for example of NCHAR/NVARCHAR handling).
 
         :param type_: Column type returned by inspector
         :return: SqlAlchemy column type
         """
+        for regex, sqla_type in cls.column_type_mappings:
+            match = regex.match(type_)
```

Comment:
@villebro we tested this during a deploy and got errors here because sometimes type_ is  None. get_sqla_column_type is used in `connectors/sqla/models.py` where sometimes the type on a column can be null.